### PR TITLE
[tests-only][full-ci] removing usage of a stepdefination inside another stepdefination in WebDavPropertiesContext

### PR DESCRIPTION
**Pull request by ghost**
## Description

removing the use of a stepdefinition inside another stepdefinition in `WebDavPropertiesContext.php`

## Related Issue

- Related issue: https://github.com/owncloud/ocis/issues/8199

## How Has This Been Tested?

- test environment:
- locally
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/FavoritesContext.php [hunk 1]
```diff
@@ -252,10 +252,11 @@ public function theUserHasUnfavoritedElementUsingWebDavApi(string $path):void {
 	 */
 	public function asUserFileOrFolderShouldBeFavorited(string $user, string $path, int $expectedValue = 1):void {
 		$property = "oc:favorite";
-		$this->webDavPropertiesContext->asUserFolderShouldContainAPropertyWithValue(
+		$this->webDavPropertiesContext->checkPropertyOfAFolder(
 			$user,
 			$path,
 			$property,
+			(string)$expectedValue,
 			(string)$expectedValue
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FavoritesContext.php:260]:
```suggestion
			(string)$expectedValue
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -3495,7 +3495,7 @@ public function userGetsTheFollowingPropertiesOfFileInsideSpaceWithValueUsingThe
 		string $value
 	):void {
 		$this->setSpaceIDByName($user, $spaceName);
-		$this->webDavPropertiesContext->asUserFolderShouldContainAPropertyWithValue($user, $resourceName, $property, $value);
+		$this->webDavPropertiesContext->checkPropertyOfAFolder($user, $resourceName, $property, $value, $value);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3498]:
```suggestion
		$this->webDavPropertiesContext->checkPropertyOfAFolder($user, $resourceName, $property, $value);
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 3]
```diff
@@ -912,45 +923,41 @@ public function asUserFolderShouldContainAPropertyWithValueOrWithValue(
...
-	 * @throws Exception
 	 */
-	public function asUserFolderShouldContainAPropertyWithValue(
+	public function checkPropertyOfAFolder(
 		string $user,
 		string $path,
 		string $property,
-		string $value
+		string $expectedValue,
+		?string $altExpectedValue
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:945]:
```suggestion
		?string $altExpectedValue = null
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 4]
```diff
@@ -616,6 +615,17 @@ public function valueOfItemOfPathShouldBe(string $user, string $xpath, string $p
 	 * @return void
 	 * @throws Exception
 	 */
+	public function theValueOfTheItemInTheResponseAboutUserShouldBe(string $xpath, ?string $user, string $expectedValue):void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:618]: also remove null from code docs
```suggestion
	public function theValueOfTheItemInTheResponseAboutUserShouldBe(string $xpath, string $user, string $expectedValue):void {
```

**Review by saw-jan — APPROVED** [review]

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

**Pull request merged**